### ISSUE TITLE
Create shapes on long-hold pen.

### ISSUE THREAD

**Issue by dotancohen** [olup/notable#25]
The Samsung Note device has a terrific feature for drawing shapes. With the stylus, one can draw e.g. a circle and at the end of the stroke not lift the stylus. The program will then change the hand-drawn circle to a vector that approximates the hand-drawn shape. This works with circles, rectangles, squares, stars, and other shapes.

The ability to convert hand-drawn shapes to vectors if the stylus is not raised would be a terrific addition to Notable.

Example video:
https://www.youtube.com/watch?v=LD_YHwCF_ZU

**Comment by olup** [comment]
I agree, and this was one of the original features I wanted to add. I have been considering different paths to get there, but nothing has stood out as a perfect way.
In my view it either uses a simple algo to straighten lines or a complex machine learning system to detect the shape. But either way I haven't worked on an implementation.

If you know some code that does it, whatever the language, that could be a good start to implement the feature

**Comment by dotancohen** [comment]
Googling for [algorithm to create circles from hand-drawn shapes](https://www.google.com/search?q=algorithm+to+create+circles+from+hand-drawn+shapes) yielded a few results, most notably this SE question on [[Recognition of handwritten circles, diamonds and rectangles](https://stackoverflow.com/questions/20254440/recognition-of-handwritten-circles-diamonds-and-rectangles)](https://stackoverflow.com/questions/20254440/recognition-of-handwritten-circles-diamonds-and-rectangles).
